src/main/java/edu/kis/powp/jobs2d/command/CompoundCommand.java

```diff
@@ -50,6 +51,39 @@ public void clearCommand() {
         commands.clear();
     }
 
+    public void interchangeCommands(int commandIndex1, int commandIndex2) {
+        if (commands.isEmpty()) {
+            throw new IndexOutOfBoundsException("Compound command is empty");
+        }
+
+        if (commandIndex1 >= commands.size() || commandIndex1 < 0 || commandIndex2 >= commands.size() || commandIndex2 < 0) {
+            throw new IndexOutOfBoundsException("Index out of bounds");
+        }
+
+        DriverCommand temp = commands.get(commandIndex1);
+
+        commands.set(commandIndex1, commands.get(commandIndex2));
+        commands.set(commandIndex2, temp);
+    }
+
+    public void modifyCoordinates(int commandIndex, int x, int y) {
```

Comment:
It would be better to avoid mutating command (immutable approach) and create separate class to handle modifications (e.g., through generation of new commands) and be more SRP